open_graph.py

```diff
@@ -29,9 +30,13 @@ def tag_article(instance):
     ogtags = [('og:title', instance.title),
               ('og:type', 'article')]
 
+    soup = BeautifulSoup(instance._content, 'html.parser')
+    img_links = soup.find_all('img')
     image = instance.metadata.get('og_image', '')
     if image:
         ogtags.append(('og:image', image))
+    elif (len(img_links) > 0):
+        ogtags.append(('og:image', img_links[0].get('src')))
 
```

Comment:
Thanks. Two comments:

1. It would be more efficient to do the html parsing (lines 33-34) inside the else.

2. Image paths are often relatives. We should ensure that the URL is absolute. OG does not resolve relative URL. 

Something like this should work:

    for f in settings['filenames']:
        if os.path.basename(img['src']) in f:
            url = os.path.join(settings['OUTPUT_PATH'], settings['filenames'][f].save_as)
    else:
        base_path = os.path.join(settings['OUTPUT_PATH'], img['src'][1:])

This code (taken from the `image_process` plugin) should handle file paths that uses the ``{attached}`` syntax. If you know something better (like a call to pelican to do the conversion) I would be happy to know about it.